src/sentry/api/endpoints/project_key_stats.py

```diff
@@ -30,35 +39,68 @@ def get(self, request: Request, project, key_id) -> Response:
         except ProjectKey.DoesNotExist:
             raise ResourceDoesNotExist
 
+        # Outcomes queries are coupled to Django's QueryDict :(
+        query_data = QueryDict(mutable=True)
+        query_data.setlist("field", ["sum(quantity)"])
+        query_data.setlist(
+            "outcome",
+            [
+                Outcome.ACCEPTED.api_name(),
+                Outcome.FILTERED.api_name(),
+                Outcome.RATE_LIMITED.api_name(),
+            ],
+        )
+        query_data["groupBy"] = "outcome"
+        query_data["category"] = "error"
+
+        now = timezone.now()
+        query_data["end"] = request.GET.get("until", now.isoformat())
+        query_data["start"] = request.GET.get("since", (now - timedelta(days=30)).isoformat())
+        query_data["interval"] = request.GET.get("resolution", "1d")
```

Comment:
Similarly, here we used to compute the "optimal" rollup based on the provided start and end time:

https://github.com/getsentry/sentry/blob/4888bf6769fe25d30f4296c3b06566564313e745/src/sentry/tsdb/base.py#L214-L237

Switching this to a static 1-day interval should work though, since the frontend statically requests `1d always: https://github.com/getsentry/sentry/blob/bb42837000875ecebc4276a0d6b4a68b30375eb4/static/app/views/settings/project/projectKeys/details/keyStats.tsx#L60-L62

Reply:
The outcomes query interface has a much less flexible interval range unfortunately. The current changes *could* break an API integration that is doing custom interval queries.